src/logger.ts

```diff
@@ -0,0 +1,145 @@
+import {
```

Comment:
I feel like node or a simple dependency can provide this logic (a logger)

may be worth looking into